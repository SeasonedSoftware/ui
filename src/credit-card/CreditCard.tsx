import React, { useState } from 'react'
import { Grid } from '@material-ui/core'

import FormFields from './FormFields'
import Card from './Card'

export const CreditCard = (props:any) => {
  const [focused, setFocused] = useState()

  const handleFocus = ({ target: { id } }:any) => {
    setFocused(id)
  }

  return (
    <>
      <Card {...props} focused={focused} />
      <Grid container justify={'space-between'} style={{ margin: '30px 0' }}>
        <FormFields {...props} handleFocus={handleFocus} />
      </Grid>
    </>
  )
}

export default CreditCard
