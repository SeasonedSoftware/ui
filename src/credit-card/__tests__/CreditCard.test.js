import React from 'react'
import renderer from 'react-test-renderer'

import CreditCard from '../CreditCard'

jest.mock('../FormFields', () => props => <div {...props}>FormFields</div>)
jest.mock('../Card', () => props => <div {...props}>Card</div>)

it('renders correctly', () => {
  const props = {
    foo: 'bar',
  }
  const tree = renderer.create(<CreditCard {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
