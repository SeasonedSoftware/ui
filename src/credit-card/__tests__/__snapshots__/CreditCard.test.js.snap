// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    foo="bar"
  >
    Card
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
    style={
      Object {
        "margin": "30px 0",
      }
    }
  >
    <div
      foo="bar"
      handleFocus={[Function]}
    >
      FormFields
    </div>
  </div>,
]
`;
